### DISCUSSION TITLE
Feature idea  - Workflow file dependencies

### DISCUSSION THREAD

**Discussion by racklking** [jhuckaby/Cronicle#575] [Ideas]
First of all, amazing job with Chronicle and looking desperately for Orchestra.
I have suggestion for a feature.

I sometimes need to refer to files for a previous step.
For example, I have two tasks in my workflow , Task1 and Task2
Task1 writes a file, T1_run1_attempt1
I refer this file in Task2

If I execute Task1 again for some reason, I need to refer the latest file, that is
T1_run1_attempt2

Basically, the Task2 should automatically know which is the latest file from Task1.

So if the tasks can have a standard file prefix name attached to them, based on the run number and attempt number, it would be great. Plus a standard way to automatically refer to the latest file of a Task.

Thanks again for the amazing job !

**Comment by jhuckaby** [comment]
Thank you!  And yes, this feature is implemented in Orchestra.  Jobs can "upload" files very easily, and they are stored and can then be retrieved (downloaded), and used in other jobs in the same workflow, or in a chain reaction string.

The reason files need to be "uploaded" and "downloaded" is because each individual job may run on a different worker server.

**Reply by mcfetz** [reply]
This sounds like a really great feature! Would help us a lot handling files through ur flows instead of using temporary directories.
